pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -241,9 +264,25 @@ def save_checkpoint(self, trainer, pl_module):
         # Mode 2: save the last checkpoint
         self._save_last_checkpoint(trainer, monitor_candidates)
 
+    def _should_skip_saving_checkpoint(self, trainer) -> bool:
+        return (
+            trainer.fast_dev_run  # disable checkpointing with fast_dev_run
+            or trainer.running_sanity_check  # don't save anything during sanity check
+            or self.save_top_k == 0  # no models are saved
+            or self._last_global_step_saved == trainer.global_step  # already saved at the last step
```

Comment:
@tchaton this uses the global step instead of the total batch idx. should we use the same across both?